test/test_xla_backend.py

```diff
@@ -0,0 +1,442 @@
+import os
+import pytest
+import random
+import re
+import string
+import torch
+import torch.distributed as dist
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.core.xla_env_vars as xenv
+import torch_xla.distributed.xla_backend
+
+from contextlib import contextmanager
+from datetime import timedelta
+from pathlib import Path
+
+os.environ[xenv.WORLD_SIZE] = '1'
+os.environ[xenv.ORDINAL] = '0'
+os.environ[
```

Comment:
These tests are not running on real devices, so I suppose we should not need to worry about the actual values of these env vars?

Reply:
We implement the test in a way that it can be run against different platform. I understand that it is a unit test which is why I think it is OK to overwrite the `requires_intercore_reduce` below. However there might be device dependent logic like generate the `all_reduce` differently if it is on `TPU` instead of `CPU`. 

I think it is better to leave the caller of the test to decide which platform it needs to run on. Test should pass as long as `world_size` is overwritten. (If not then there is something we need to fix)